apps/block_scout_web/assets/js/omni.js

```diff
@@ -1,199 +1,318 @@
+    .join(' ')
 
 // default api indexerUrl point to local server
-let indexerUrl = "http://localhost:8888";
+// let indexerUrl = 'http://localhost:8888'
+// in dev, use testnet explorer
+let indexerUrl = 'https://testnet-xapi.explorer.omni.network'
 
 const xchainMarker = () => {
-  const div = document.createElement("div");
-  div.style =
-    "height: inherit;margin-left: 1rem;background-color: #8258cd;padding: 0 0.5rem;border-radius: 1rem;";
-  div.innerHTML =
-    '<img style="height: 1.5rem;display: inline-block;margin-top: -0.1rem" src="/images/favicon-32x32.png"> <div style="margin-top: 0.2rem;display: inline-block;font-weight: bolder;color: white;">X-Rollup</div>';
-  return div;
-};
-
-const customXChainDataView = (details) => {
-  for (let i = 0; i < details.length; i++) {
-    let externalChain = "";
-    if (details[i]["SourceChain"] != "omni") {
-      externalChain = details[i]["SourceChain"];
-    } else if (details[i]["DestinationChain"] != "omni") {
-      externalChain = details[i]["DestinationChain"];
-    }
-    let externalChainExplorer = "";
-    if (explorerMap[externalChain] !== undefined) {
-      externalChainExplorer = explorerMap[externalChain];
-    }
-    details[i]["ChainTxHash"] =
-      externalChainExplorer + details[i]["ChainTxHash"];
+  const div = document.createElement('div')
+
+  div.className = 'xchain-marker'
+  div.innerHTML = `
+    <img class="xchain-marker__logo" src="/images/omni_logo.svg" />
+    <p class="xchain-marker__label">X-Rollup</p>
+  `
+  return div
+}
+
+const infoIcon = `<i class="xchain-data__detail-icon fa-solid fa-circle-info"></i>`
+const successIcon = `<i style="color: #20b760;" class="fa-regular fa-circle-check"></i>`
+const errorIcon = `<i style="color: #dc3545;" class="fa-solid fa-circle-exclamation"></i>`
+
+const xchainDetail = (label, value) => `
+  <div class="xchain-data__detail">
+    ${infoIcon}
+    <p class="xchain-data__detail-label">
+      ${label}
+    </p>
+    <p class="xchain-data__detail-value">
+      ${value}
+    </p>
+  </div>
+`
+
+const xchainDetailLink = (label, value, link) => `
+  <div class="xchain-data__detail">
+    ${infoIcon}
+    <p class="xchain-data__detail-label">
+      ${label}
+    </p>
+    <a
+      class="xchain-data__detail-value"
+      href="${link}"
+      target="_blank"
+      rel="noopener noreferrer"
+    >
+      ${value}
+    </a>
+  </div>
+`
+
+const xchainSuccess = `<span class="success">${successIcon} Success</span>`
+const xchainError = `<span class="error">${errorIcon} Error</span>`
+
+const xchainDetails = details => {
+  const div = document.createElement('div')
+
+  const {
+    SourceChain,
+    DestinationChain,
+    OmniTxHash,
+    ChainTxHash,
+    ChainTxStatus,
+    SourceNonce,
+    ChainBlockNumber,
+    ExternalChainDisplayName,
+    ExternalTxLink,
+  } = details
+
+  div.className = 'xchain-data__details'
+
+  div.innerHTML += xchainDetail('Source Chain', chainDisplayName(SourceChain))
```

Comment:
are there some implications with changing innerHTML bunch of times? maybe we could prepare innerHTML in some variable and just set it once instead of appending directly to innerHTML?

Reply:
not in terms of like screen flashing a bunch of time, no. new innerHtml won't be painted until current synchronous block of js finishes (so this function). but maybe there are some other unexpected things. so sure let's do that